substrate/frame/support/src/traits/tokens/fungible/mod.rs

```diff
@@ -17,26 +17,129 @@
 //!
-//! ### User-implememted traits
-//! - `Inspect`: Regular balance inspector functions.
-//! - `Unbalanced`: Low-level balance mutating functions. Does not guarantee proper book-keeping and
-//!   so should not be called into directly from application code. Other traits depend on this and
-//!   provide default implementations based on it.
-//! - `UnbalancedHold`: Low-level balance mutating functions for balances placed on hold. Does not
+//! Also see the [`frame_tokens`] reference docs for more information about the place of
+//! `fungible` traits in Substrate.
+//!
+//! # Avaliable Traits
+//! - [`Inspect`]: Regular balance inspector functions.
+//! - [`Unbalanced`]: Low-level balance mutating functions. Does not guarantee proper book-keeping
+//!   and so should not be called into directly from application code. Other traits depend on this
+//!   and provide default implementations based on it.
+//! - [`UnbalancedHold`]: Low-level balance mutating functions for balances placed on hold. Does not
 //!   guarantee proper book-keeping and so should not be called into directly from application code.
 //!   Other traits depend on this and provide default implementations based on it.
-//! - `Mutate`: Regular balance mutator functions. Pre-implemented using `Unbalanced`, though the
-//!   `done_*` functions should likely be reimplemented in case you want to do something following
-//!   the operation such as emit events.
-//! - `InspectHold`: Inspector functions for balances on hold.
-//! - `MutateHold`: Mutator functions for balances on hold. Mostly pre-implemented using
-//!   `UnbalancedHold`.
-//! - `InspectFreeze`: Inspector functions for frozen balance.
-//! - `MutateFreeze`: Mutator functions for frozen balance.
-//! - `Balanced`: One-sided mutator functions for regular balances, which return imbalance objects
+//! - [`Mutate`]: Regular balance mutator functions. Pre-implemented using [`Unbalanced`], though
+//!   the `done_*` functions should likely be reimplemented in case you want to do something
+//!   following the operation such as emit events.
+//! - [`InspectHold`]: Inspector functions for balances on hold.
+//! - [`MutateHold`]: Mutator functions for balances on hold. Mostly pre-implemented using
+//!   [`UnbalancedHold`].
+//! - [`InspectFreeze`]: Inspector functions for frozen balance.
+//! - [`MutateFreeze`]: Mutator functions for frozen balance.
+//! - [`Balanced`]: One-sided mutator functions for regular balances, which return imbalance objects
 //!   which guarantee eventual book-keeping. May be useful for some sophisticated operations where
 //!   funds must be removed from an account before it is known precisely what should be done with
 //!   them.
+//!
+//! ## Terminology
+//!
+//! - **Total Issuance**: The total number of units in existence in a system.
+//!
+//! - **Total Balance**: The sum of an account's free and held balances.
+//!
+//! - **Free Balance**: A portion of an account's total balance that is not held.
```

Comment:
I have two comments about this: 

1. Not sure if it is correct, as intuitively I would think that your free balance is what you can tranafer, which should also take freezes into account. 
2. I don't see any corresponding function in the new traits, so not sure if the definition is useful. It seems more like a term that is useful in `Currency` era, not so much in `fungible::*`. 

I think think that a good "transferrable" definition would be useful for wallets, and they already reverse engineer it.

Reply:
I was confused about this too. The terminology is used frequently enough in trait impls that I think it's worth mentioning so there's a single source of truth.

As explained by Gav [here](https://github.com/paritytech/polkadot-sdk/issues/1833#issuecomment-1805764506), `free` is `total - held`. This is different to `spendable` balance, which takes `freezes` into account. 

This is described in the `Visualising Balance Components Together` section, but I see how it could be confusing, and will add a dot point for `Spendable` in the `Terminology` section and some clarification in each's description.